### PR TITLE
feat: implement MinCollateralLiquidity

### PR DESCRIPTION
**Pull request by toteki**
## Description

closes: #1163 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by toteki** [comment]
Note: Merge #1329 then #1330 before this

**Review by khoerling — APPROVED** [review]
Looks solid, haven't pulled this down to run locally yet...

File: x/leverage/keeper/keeper.go [hunk 1]
```diff
@@ -197,6 +197,11 @@ func (k Keeper) Withdraw(ctx sdk.Context, supplierAddr sdk.AccAddress, uToken sd
 		return sdk.Coin{}, err
 	}
 
+	// check MinCollateralLiquidity is still satisfied after the transaction
+	if err = k.checkCollateralLiquidity(ctx, token.Denom); err != nil {
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/leverage/keeper/keeper.go:201]: Not for the collateral liquidity requirement - it prevents `MsgCollateralize`, `MsgBorrow`, and `MsgWithdraw` to protect the availability of `MsgLiquidate`.

**Review by RafilxTenfen — APPROVED** [review]

File: x/leverage/keeper/keeper.go [hunk 2]
```diff
@@ -257,7 +262,9 @@ func (k Keeper) Borrow(ctx sdk.Context, borrowerAddr sdk.AccAddress, borrow sdk.
 	if utilization.GT(token.MaxSupplyUtilization) {
 		return types.ErrMaxSupplyUtilization.Wrap(utilization.String())
 	}
-	return nil
+
+	// check MinCollateralLiquidity is still satisfied after the transaction
+	return k.checkCollateralLiquidity(ctx, borrow.Denom)
```
Comment (toteki) [inline x/leverage/keeper/keeper.go:267]:
```suggestion
	return k.checkCollateralLiquidity(ctx, types.ToUTokenDenom(borrow.Denom))
```

**Pull request merged**